apollo-backend/src/main/java/io/logz/apollo/transformers/ingress/IngressLabelTransformer.java

```diff
@@ -0,0 +1,40 @@
+package io.logz.apollo.transformers.ingress;
+
+import io.fabric8.kubernetes.api.model.extensions.Ingress;
+import com.google.common.collect.ImmutableMap;
+import io.logz.apollo.kubernetes.ApolloToKubernetes;
+import io.logz.apollo.transformers.LabelsNormalizer;
+
+import java.util.Map;
+import java.util.Optional;
+
+
+public class IngressLabelTransformer implements BaseIngressTransformer {
+    @Override
+    public Ingress transform(Ingress ingress,
+                             io.logz.apollo.models.Deployment apolloDeployment,
+                             io.logz.apollo.models.Service apolloService,
+                             io.logz.apollo.models.Environment apolloEnvironment) {
+
+        Map<String, String> desiredLabels = ImmutableMap.<String, String> builder()
+                .put("environment", LabelsNormalizer.normalize(apolloEnvironment.getName()))
+                .put("geo_region", LabelsNormalizer.normalize(apolloEnvironment.getGeoRegion()))
+                .put("apollo_unique_identifier", ApolloToKubernetes.getApolloServiceUniqueIdentifier(apolloEnvironment,
+                        apolloService, Optional.empty()))
```

Comment:
What if there is a group name? This unique identifier would be wrong...

Reply:
not sure i follow, why the group name should effect the ingress rule unique identifier ?